### PR TITLE
#164798922 Implement article share via mail functionality

### PR DESCRIPTION
**Pull request by chukwudidu**
#### Description
Creates the share article via mail functionality.
#### Type of change
- [x]  Non-breaking change(introduces new endpoint to codebase)
#### How Has This Been Tested?
- All tests are currently passing
- It was also tested on Postman

##### Screenshots
Screenshot 2019-04-14 at 9 28 03 PM
Screenshot 2019-04-14 at 9 27 31 PM

#### Checklist:
- [x] Add unit tests for endpoint
- [x] Add share endpoint
- [x] Add documentation for endpoint
#### PIVOTAL BOARD
[#164798922](https://www.pivotaltracker.com/story/show/164798922)

### CODE REVIEW

**Review by TheDrizzyWay — APPROVED** [review]
Good job.

File: server/controllers/article.controller.js [hunk 1]
```diff
@@ -166,5 +166,43 @@ class ArticleController {
     }
     return Response(res, 200, 'Articles successfully retrieved', { articles });
   }
+
+  /**
+   * Share article controller
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} response
```
Comment (geekdanitek) [inline server/controllers/article.controller.js:174]: The `returns` description in JSDOCS can be more descriptive

File: server/controllers/article.controller.js [hunk 2]
```diff
@@ -166,5 +166,43 @@ class ArticleController {
...
+   */
+  static async shareArticle(req, res) {
+    const {
+      body: { email },
```
Comment (geekdanitek) [inline server/controllers/article.controller.js:178]: Nice work on using nested destructing

**Review by geekdanitek — APPROVED** [review]

File: server/controllers/article.controller.js (hunk 1 above)
Comment (davidokonji) [reply server/controllers/article.controller.js:174]: @geekdanitek Yes, the value returned is actually the response data

Comment (geekdanitek) [reply server/controllers/article.controller.js:174]: Ok but it would be nice if there is a description of the return data

**Review by davidokonji — APPROVED** [review]
Good work.

**Comment by Chiazokam** [comment]
Great work here

**Review by Chiazokam — APPROVED** [review]

**Review by paradisekelechi — APPROVED** [review]

**Pull request merged**